lib/ecmascript.ts

```diff
@@ -4733,7 +4755,9 @@ export function RoundDuration(
 
   // First convert time units up to days, if rounding to days or higher units.
   // If rounding relative to a ZonedDateTime, then some days may not be 24h.
-  let dayLengthNs: JSBI;
+  // TS doesn't know that `dayLengthNs` is only used if the unit is day or
+  // larger. This makes the cast below acceptable.
+  let dayLengthNs: JSBI = undefined as unknown as JSBI;
```

Comment:
I'm not a good judge of what's idiomatic, but might it be better to type this as `JSBI | undefined` and use `!` in the branch where we know it is not undefined?

Reply:
I think I'd type this as `dayLengthNs: JSBI|undefined` and then cast using `!` along with a comment explaining why we know for certain it's been assigned a value there. This might force us to write this out multiple times, but I think that it's best to document these assumptions.

Reply:
OK, I checked in @12wrigja's suggestion.